lametro/views.py

```diff
@@ -142,110 +150,56 @@ def get_context_data(self, **kwargs):
-                # rather than get().
-                board_report = LAMetroBill.objects.filter(ocd_id=ocd_id)
-                if board_report:
-                    obj = obj + (board_report.first().inferred_status,)
-                    cursor_copy.append(obj)
-
-            # Create a named tuple
-            board_report_tuple = namedtuple('BoardReportProperties', columns, rename=True)
-            # Put results inside a list with assigned fields (from namedtuple)
-            related_board_reports = [board_report_tuple(*r) for r in cursor_copy]
-
-            # Find agenda link.
-            if event.documents.all():
-                for document in event.documents.all():
-                    if "Agenda" in document.note:
-                        context['agenda_url'] = document.url
-                        context['document_timestamp'] = document.updated_at
-                    elif "Manual upload URL" in document.note:
-                        context['uploaded_agenda_url'] = document.url
-                        context['document_timestamp'] = document.updated_at
-                    elif "Manual upload PDF" in document.note:
-                        context['uploaded_agenda_pdf'] = document.url
-                        context['document_timestamp'] = document.updated_at
-                        '''
-                        LA Metro Councilmatic uses the adv_cache library to partially cache templates: in the event view, we cache the entire template, except the iframe. (N.B. With this library, the views do not cached, unless explicitly wrapped in a django cache decorator.
-                        Nonetheless, several popular browsers (e.g., Chrome and Firefox) retrieve cached iframe images, regardless of the site's caching specifications.
-                        We use the agenda's "updated_at" timestamp to bust the iframe cache: we save it inside context and then assign it as the "name" of the iframe, preventing the browser from retrieving a cached iframe, when the timestamp changes.
-                        '''
-
-            context['related_board_reports'] = related_board_reports
-            context['base_url'] = PIC_BASE_URL # Give JS access to this variable
-
-            # Render forms if not a POST request
-            if 'url_form' not in context:
-                context['url_form'] = AgendaUrlForm()
-            if 'pdf_form' not in context:
-                context['pdf_form'] = AgendaPdfForm()
+            context['minutes'] = event.documents.get(note__icontains='minutes')
+        except EventDocument.DoesNotExist:
+            pass
+
+        agenda_with_board_reports = event.agenda\
+            .filter(related_entities__bill__versions__isnull=False)\
+            .annotate(int_order=Cast('order', IntegerField()))\
+            .order_by('int_order')
+        
+        # Find agenda link.
+        if event.documents.all():
+            for document in event.documents.all():
+                if "Agenda" in document.note:
```

Comment:
Should we break after one of these conditions is met?

Reply:
yup, good idea.

Reply:
actually, I'm not sure. the current code doesn't do that. seems like we might want to have more than once of these in the view context.